tests/components/linkplay/test_config_flow.py

```diff
@@ -0,0 +1,186 @@
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == SOURCE_USER
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {CONF_HOST: HOST},
+    )
+
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == NAME
+    assert result["data"] == {
+        CONF_HOST: HOST,
+    }
+    assert result["result"].unique_id == UUID
+
+
+async def test_user_flow_re_entry(
+    hass: HomeAssistant,
+    mock_linkplay_factory_bridge: AsyncMock,
+    mock_setup_entry: AsyncMock,
+) -> None:
+    """Test user setup config flow."""
+
+    # Create mock entry which already has the same UUID
+    entry = MockConfigEntry(
+        data={CONF_HOST: HOST},
+        domain=DOMAIN,
+        title=NAME,
+        unique_id=UUID,
+        source=SOURCE_USER,
+    )
+    entry.add_to_hass(hass)
+
+    # Re-create entry with different host
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+    )
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {CONF_HOST: HOST_REENTRY},
+    )
+
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_zeroconf_flow(
+    hass: HomeAssistant,
+    mock_linkplay_factory_bridge: AsyncMock,
+    mock_setup_entry: AsyncMock,
+) -> None:
+    """Test Zeroconf flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_ZEROCONF},
+        data=ZEROCONF_DISCOVERY,
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "discovery_confirm"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {},
+    )
+
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == NAME
+    assert result["data"] == {
+        CONF_HOST: HOST,
+    }
+    assert result["result"].unique_id == UUID
+
+
+async def test_zeroconf_flow_re_entry(
+    hass: HomeAssistant,
+    mock_linkplay_factory_bridge: AsyncMock,
+    mock_setup_entry: AsyncMock,
+) -> None:
+    """Test Zeroconf flow."""
+
+    # Create mock entry which already has the same UUID
+    entry = MockConfigEntry(
+        data={CONF_HOST: HOST},
+        domain=DOMAIN,
+        title=NAME,
+        unique_id=UUID,
+        source=SOURCE_ZEROCONF,
+    )
+    entry.add_to_hass(hass)
+
+    # Re-create entry with different host
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_ZEROCONF},
+        data=ZEROCONF_DISCOVERY_RE_ENTRY,
+    )
+
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"
+
+
+async def test_flow_errors(
```

Comment:
Can we make this test so that it succeeds in the end so we also test that the flow is able to recover